pkg/controller/servingcert/controller/secret_creating_controller.go

```diff
@@ -301,23 +318,34 @@ func toBaseSecret(service *corev1.Service) *corev1.Secret {
 	}
 }
 
-func getServingCert(dnsSuffix string, ca *crypto.CA, service *corev1.Service) (*crypto.TLSCertificateConfig, error) {
-	dnsName := service.Name + "." + service.Namespace + ".svc"
+func GetServingCert(dnsSuffix string, ca *crypto.CA, intermediateCACert *x509.Certificate, serviceObjectMeta *metav1.ObjectMeta) (*crypto.TLSCertificateConfig, error) {
```

Comment:
this is not a getter, but a constructor. NewServingCert

Reply:
or MakeServingCert

Reply:
Done